Modules/TPC/src/PadCalibrationCheck2D.cxx

```diff
@@ -0,0 +1,144 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+///
+/// \file   PadCalibrationCheck2D.cxx
+/// \author Laura Serksnyte
+///
+
+#include "TPC/PadCalibrationCheck2D.h"
+#include "QualityControl/MonitorObject.h"
+#include "QualityControl/Quality.h"
+
+#include <fairlogger/Logger.h>
+// ROOT
+#include <TCanvas.h>
+#include <TH2.h>
+#include <TPad.h>
+#include <TList.h>
+#include <TPaveText.h>
+
+using namespace std;
+
+namespace o2::quality_control_modules::tpc
+{
+
+void PadCalibrationCheck2D::configure(std::string) {}
+Quality PadCalibrationCheck2D::check(std::map<std::string, std::shared_ptr<MonitorObject>>* moMap)
+{
+
+  auto mo = moMap->begin()->second;
+  Quality result = Quality::Null;
+  if (mo->getName() == "c_ROCs_Pedestal_2D") {
+    result = Quality::Good;
+    auto* canv = dynamic_cast<TCanvas*>(mo->getObject());
```

Comment:
dynamic cast only makes sense if you want the verify the object type. So I guess you should either change this to `static_cast` of c-style cast `(TCanvas*)` or add
```c++
if (!canv) {
  return Quality::Null;
}